features/step_definitions/barclamp_pacemaker/pacemaker_steps.rb

```diff
@@ -0,0 +1,292 @@
+        expect(status).to eq("complete")
+    end
+end
+
+And(/^I can update and delete new attribute to a crm node$/) do
+    @primary_node.exec!("crm_attribute --node d52-54-01-77-77-01 --name location --update office")
+    @primary_node.exec!("crm_attribute --node d52-54-01-77-77-01 --name location --query")
+    @primary_node.exec!("crm_attribute --node d52-54-01-77-77-01 --name location --delete")
+end
+
+And(/^I can verify the status of corosync being active with no faults and correct ring addresses$/) do
+    prim_ring_addr = ""
+    prim_status = ""
+    sec_ring_addr = ""
+    sec_status = ""
+   
+    # Get ring address from node configuration
+    response = @primary_node.exec!("crm corosync get nodelist.node.ring0_addr").output
+    ring_addr_conf = response.split(/\n/)
+    puts "Ring Addresses from configure: #{ring_addr_conf}"
+
+    # Get ring status from primary node
+    response = @primary_node.exec!("crm corosync status").output
+    puts "Primary Node Corosync: #{response}"
+    lines = response.tr("\t", " ").split(/\n/)
+    lines.each do |line|
+        if line.include?("id =")
+            prim_ring_addr = line.split(/=/)[-1].strip!
+        end
+        if line.include?("status =")
+            prim_status = line.split(/=/)[-1].strip!
+        end
+    end
+
+    # Get ring status from secondary node
+    response = @secondary_node.exec!("crm corosync status").output
+    puts "Secondary Node Corosync: #{response}"
+    lines = response.tr("\t", " ").split(/\n/)
+    lines.each do |line|
+        if line.include?("id =")
+            sec_ring_addr = line.split(/=/)[-1].strip!
+        end
+        if line.include?("status =")
+            sec_status = line.split(/=/)[-1].strip!
+        end
+    end
+
+    expect(ring_addr_conf).to include(prim_ring_addr, sec_ring_addr)
+    expect(prim_status).to eq("ring 0 active with no faults")
+    expect(sec_status).to eq("ring 0 active with no faults")
+end
+
+And(/^I can move a cluster node to standby and back to online without affecting resources$/) do
+    @primary_node.exec!("crm node standby")
+    response = @primary_node.exec!("crm_mon -1").output
+    lines = response.split(/\n/)
+    puts "CRM Monitor Standby: #{response}"
+    online_nodes = []
+    standby_nodes = []
+    lines.each do |line|
+        if line.include?("Standby")
+            puts "Standby line #{line}"
+            standby_nodes = line.split(/:/)[-1].delete("[]").split(/ /).reject{ |c| c.empty?}
+        elsif line.include?("Online:")
+            puts "Online line #{line}"
+            online_nodes = line.split(/:/)[-1].delete("[]").split(/ /).reject{ |c| c.empty?}
+        end
+    end
+    expect(online_nodes).to include(@secondary_node.attributes()[:name])
+    expect(standby_nodes).to include(@primary_node.attributes()[:name])
+
+    @primary_node.exec!("crm node online")
+    response = @primary_node.exec!("crm_mon -1").output
+    lines = response.split(/\n/)
+    standby_nodes = []
```

Comment:
Again this looks like there is quite a lot of duplication.  Could you refactor into a helper method here too?